api/python/cellxgene_census/src/cellxgene_census/experimental/ml/geneformer_tokenizer.py

```diff
@@ -0,0 +1,175 @@
+
+    Example usage:
+
+    ```
+    import cellxgene_census
+    import tiledbsoma
+    from cellxgene_census.experimental.ml import GeneformerTokenizer
+
+    with cellxgene_census.open_soma() as census:
+        with GeneformerTokenizer(
+            census["census_data"]["homo_sapiens"],
+            # set obs_query to define some subset of Census cells:
+            obs_query=tiledbsoma.AxisQuery(value_filter="is_primary_data == True and tissue_general == 'tongue'"),
+            obs_attributes=(
+                "soma_joinid",
+                "cell_type_ontology_term_id",
+            ),
+        ) as tokenizer:
+            dataset = tokenizer.build()
+    ```
+
+    Dataset item contents:
+    - `input_ids`: Geneformer token sequence for the cell
+    - `length`: Length of the token sequence
+    - and the specified `obs_attributes` (cell metadata from the experiment obs dataframe)
+    """
+
+    obs_attributes: Set[str]
+    max_input_tokens: int
+
+    # set of gene soma_joinids corresponding to genes modeled by Geneformer:
+    model_gene_ids: npt.NDArray[np.int64]
+    model_gene_tokens: npt.NDArray[np.int64]  # token for each model_gene_id
+    model_gene_medians: npt.NDArray[np.float64]  # float for each model_gene_id
+
+    def __init__(
+        self,
+        experiment: tiledbsoma.Experiment,
+        *,
+        obs_attributes: Optional[Sequence[str]] = None,
+        max_input_tokens: int = 2048,
+        token_dictionary_file: str = "",
+        gene_median_file: str = "",
+        **kwargs: Any,
+    ) -> None:
+        """
+        - `experiment`: Census Experiment to query
+        - `obs_query`: obs AxisQuery defining the set of Census cells to process (default all)
+        - `obs_attributes`: names of attributes to propagate from the experiment obs dataframe
+           (cell metadata) into each Dataset item
+        - `max_input_tokens`: maximum length of Geneformer input token sequence (default 2048)
+        - `token_dictionary_file`, `gene_median_file`: pickle files supplying the mapping of
+          Ensembl human gene IDs onto Geneformer token numbers and median expression values.
+          By default, these will be loaded from the Geneformer package.
+        """
+        assert (
+            "normalized" in experiment.ms["RNA"].X
+        ), "Experiment RNA measurement lacks 'normalized' layer; try 'latest' Census version (2023-08-01 or newer)"
+
+        self.max_input_tokens = max_input_tokens
+        self.obs_attributes = set(obs_attributes) if obs_attributes else set()
+        self.load_geneformer_data(experiment, token_dictionary_file, gene_median_file)
+        super().__init__(
+            experiment,
+            measurement_name="RNA",
+            layer_name="normalized",
+            # configure query to fetch the relevant genes only
+            var_query=tiledbsoma.AxisQuery(coords=(self.model_gene_ids,)),
+            **kwargs,
+        )
+
+    def load_geneformer_data(
```

Comment:
nit, unless we expect users to call this manually, let's underscore it
```suggestion
    def _load_geneformer_data(
```